angular-chart.js

```diff
@@ -135,6 +136,12 @@
           aliasVar('chartHover', 'hover');
 
           // Order of setting "watch" matter
+          function setColoursListener() {
+            // Sets and returns the variable used to cancel the watch for colours, preventing the 
+            // diagram from being created due to a change occuring while creating a diagram
+            cancel_watch_for_colours = scope.$watch('colours', resetChart, true);
+            return cancel_watch_for_colours;
```

Comment:
You return `cancel_watch_for_colours` but you are not using the return value. Instead of mutating the variable in the inner function prefers setting the value of the variable from the function.

    var cancelWatchForColours = setColoursListener();